src/libslic3r/Polygon.cpp

```diff
@@ -259,6 +259,87 @@ Point Polygon::point_projection(const Point &point) const
     return proj;
 }
 
+void Polygon::fuzzy(FuzzyShape shape, int deepness)
+{
+	Point last = points.at(points.size() - 1);
+	Point last_processed = last;
+
+	double max_length = scale_(2);
+	double min_length = scale_(1);
+
+	deepness *= 3;
+
+	bool triangle_or_sawtooth;
+	switch (shape) {
+		case FuzzyShape::Triangle:
+			triangle_or_sawtooth = false;
+			break;
+		case FuzzyShape::Sawtooth:
+			triangle_or_sawtooth = true;
+			break;
+	}
+
+	double length_sum = 0;
+	Points::iterator it = points.begin();
+    while (it != points.end()) {
```

Comment:
Here you are replicating Polygon::equally_spaced_points(), however in an inefficient way due to repeated reallocation and inserting of points in the middle of a sequence. You should use it. Polygon::equally_spaced_points() may be made more efficient for sure however.